contracts/mainnet/connectors/morpho-aave-v3/main.sol

```diff
@@ -0,0 +1,681 @@
+	 * @param _setId ID stores the amount of tokens withdrawed.
+	 */
+	function withdrawOnBehalf(
+		address _tokenAddress,
+		uint256 _amount,
+		address _onBehalf,
+		address _receiver,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(_getId, _amount);
+		MORPHO_AAVE_V3.withdraw(_tokenAddress, _amt, _onBehalf, _receiver, max_iteration);
+
+		convertWethToEth(_tokenAddress == ethAddr, TokenInterface(wethAddr), _amt);
+
+		setUint(_setId, _amt);
+
+		_eventName = "LogWithdrawOnBehalf(address,uint256,address,address,uint256,uint256)";
+		_eventParam = abi.encode(
+			_tokenAddress,
+			_amt,
+			_onBehalf,
+			_receiver,
+			_getId,
+			_setId
+		);
+	}
+
+	/**
+	 * @dev Withdraw ETH/ERC20_Token.
+	 * @notice Withdraw a token from Morpho Aave.
+	 * @param _tokenAddress The address of underlying token to withdraw.(For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _amount The amount of the token (in underlying) to withdraw. (For max: `uint256(-1)`)
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens withdrawed.
+	 */
+	function withdrawWithMaxIterations(
+		address _tokenAddress,
+		uint256 _amount,
+		address _onBehalf,
+		address _receiver,
+		uint256 _maxIteration,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(_getId, _amount);
+
+		MORPHO_AAVE_V3.withdraw(_tokenAddress, _amt, _onBehalf, _receiver, _maxIteration);
+
+		convertWethToEth(_tokenAddress == ethAddr, TokenInterface(wethAddr), _amt);
+
+		setUint(_setId, _amt);
+
+		_eventName = "LogWithdrawWithMaxIterations(address,uint256,address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(
+			_tokenAddress,
+			_amt,
+			_receiver,
+			_maxIteration,
+			_getId,
+			_setId
+		);
+	}
+
+	function withdrawCollateral(
+		address _tokenAddress,
+		uint256 _amount,
+		address _receiver,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(_getId, _amount);
+
+		MORPHO_AAVE_V3.withdrawCollateral(_tokenAddress, _amt, address(this), _receiver);
+
+		convertWethToEth(_tokenAddress == ethAddr, TokenInterface(wethAddr), _amt);
+
+		setUint(_setId, _amt);
+
+		_eventName = "LogWithdrawCollateral(address,uint256,address,uint256,uint256)";
+		_eventParam = abi.encode(
+			_tokenAddress,
+			_amt,
+			_receiver,
+			_getId,
+			_setId
+		);
+	}
+
+	function withdrawCollateralOnBehalf(
+		address _tokenAddress,
+		uint256 _amount,
+		address _onBehalf,
+		address _receiver,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(_getId, _amount);
+		MORPHO_AAVE_V3.withdrawCollateral(_tokenAddress, _amt, _onBehalf, _receiver);
```

Comment:
Apply the same logics as added here - 
using `MORPHO_AAVE_V3.collateralBalance()`